servicedirectory/spec/servicedirectory_spec.rb

```diff
@@ -0,0 +1,254 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "rspec"
+require "google/cloud/service_directory"
+require_relative "../servicedirectory"
+
+describe "Service Directory API Test" do
+  ServiceDirectory = Google::Cloud::ServiceDirectory::V1beta1
+
+  def get_namespace project:, location:, namespace:
+    client = Google::Cloud::ServiceDirectory.registration_service
+    namespace_path = client.namespace_path(
+      project:   project,
+      location:  location,
+      namespace: namespace
+    )
+    client.get_namespace name: namespace_path
+  end
+
+  def get_service project:, location:, namespace:, service:
+    client = Google::Cloud::ServiceDirectory.registration_service
+    service_path = client.service_path(
+      project:   project,
+      location:  location,
+      namespace: namespace,
+      service:   service
+    )
+    client.get_service name: service_path
+  end
+
+  def get_endpoint project:, location:, namespace:, service:, endpoint:
+    client = Google::Cloud::ServiceDirectory.registration_service
+    endpoint_path = client.endpoint_path(
+      project:   project,
+      location:  location,
+      namespace: namespace,
+      service:   service,
+      endpoint:  endpoint
+    )
+    client.get_endpoint name: endpoint_path
+  end
+
+  before :each do
+    @project   = ENV["GOOGLE_CLOUD_PROJECT"]
+    @location  = "us-central1"
+    @namespace = "ruby-test-namespace-#{Time.now.to_i}"
```

Comment:
Tests on this repo run in parallel by ruby version. They will shortly be moving to the google-cloud-ruby repo which is even more parallel. I generally prefer SecureRandom.hex or SecureRandom.uuid for ensuring unique resource names

Reply:
Done.